mars/filesystem.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 1999-2018 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from gzip import GzipFile
+
+from pyarrow import LocalFileSystem as ArrowLocalFileSystem
+try:
+    import lz4, lz4.frame
+except ImportError:
+    lz4 = None
+
+from .compat import urlparse
+
+
+compressions = {
+    'gzip': lambda f: GzipFile(fileobj=f)
+}
+
+if lz4:
+    compressions['lz4'] = lz4.frame.open
```

Comment:
How about other compress types, e.g. snappy, zstd

Reply:
I will add more compress types and add tests for file system next time.